hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/CloseContainerEventHandler.java

```diff
@@ -74,14 +79,13 @@ public void onMessage(ContainerID containerID, EventPublisher publisher) {
           .getContainer(containerID);
       // Send close command to datanodes, if the container is in CLOSING state
       if (container.getState() == LifeCycleState.CLOSING) {
+        SCMCommand<?> command = new CloseContainerCommand(
+            containerID.getId(), container.getPipelineID());
+        command.setTerm(scmContext.getTerm());
```

Comment:
Question:
when you do decide to throw the NotLeaderException out rather than handling immediately after `scmContext.getTerm()`?

Reply:
`NotLeaderException` is a subclass of `IOException` which is handled in this `onMessage()` method.